src/injected/element-finder-by-path.ts

```diff
@@ -35,12 +35,14 @@ export class ElementFinderByPath {
     };
 
     public processRequest = (message: ElementFinderByPathMessage): PromiseLike<string> => {
-        if (!this.checkSyntax(message.path[0])) {
+        let element = null;
+
+        try {
+            element = this.htmlElementUtils.querySelector(message.path[0]) as HTMLElement;
+        } catch (e) {
             return Promise.reject();
```

Comment:
`Promise.reject(e)` would give the caller a little more context if something does go wrong. I think you could actually simplify this and automatically get that behavior if you wrote `processRequest` as an async function, like this:

```ts
    public processRequest = async (message: ElementFinderByPathMessage): Promise<string> => {
        // if this throws an error, it's propogated correctly without you handling it specifically
        const element = this.htmlElementUtils.querySelector(message.path[0]) as HTMLElement;

        if (element == null) {
            // You can throw new Error here with more context than the original Promise.reject()s were giving
            throw new Error(`Element with selector ${firstSelector} not found`);
        }

        // ... etc

        if (element.tagName.toLocaleLowerCase() !== 'iframe' && message.path.length === 1) {
            // In the happy path, you can simplify and avoid the `Promise.resolve` wrapper
            return element.outerHTML;
        }

        // You can still await a `PromiseLike`, so you shouldn't have to refactor frame-communicator to enable using plain Promises here
        return await this.iterateDeeperOnIframe(element, message);
    };
```